docs/post-mortems/0004-argocd-app-sync-incident.md

```diff
@@ -0,0 +1,174 @@
+    23:30 A proactive message got sendt out to all the citizens of Oslo
+2019-12-31
+    08:23 Oslonøkkelen had been downloaded 30.000 times, distributed evenly across iOS and Android, during the past 3 hours
+    09:09 DOWNTIME START - backend fails due to the increased load
+    09:14 Received Slack notifications due to an increase in 500 status codes in the backend
+    09:16 Initiated an investigation into the issue
+    09:18 EVENT START - Team creates the Slack channel #citykey-incident-backend to better cooperate on the issue
+    09:20 Manually scaling up the number of backend instances
+    09:24 Found the error. The backend fails when it tries to read an item in the database that doesn't exist. Only happens upon a users first login
+    09:28 Pull request with a fix created: http://github.com/oslokommune/something/something/pr/298
+    09:34 Pull request 298 merged and deployed to production
+    09:39 PROBLEM FIXED - Amount of 500 status code requests are decreasing
+    09:40 DOWNTIME END - Amount of 500 status codes are down to zero
+    10:10 EVENT END - Reaches criteria of 30 minutes with normal activity
+-->
+### 2022-03-07
+
+#### Incident 1
+
+1. Team booking ran `okctl upgrade` from version 0.0.82, 0.0.92 being the latest.
+2. After a successful upgrade, we noticed synchronization errors in ArgoCD.
+3. Researching these errors, we noticed that the app sync upgrade had moved all `argocd-application.yaml` files in
+    `infrastructure/applications` to `infrastructure/<relevant cluster/argocd/applications/<application name>` which
+    included `argocd-application.yaml` files from _all_ environments.
+4. After manually reverting changes and migrating them, ArgoCD once again was OK.
+
+#### Incident 2
+
+1. Team booking ran `okctl delete application -f <app manifest>` to delete an old application which hasn't been in use
+    for a while.
+2. Delete application naively purged all resources related to the old application, where one of them was the
+    namespace manifest. The old application shared namespace with new and in use application causing these applications
+    to go down.
+
+## Impact
+<!--
+Describe the consequences this had for the organization.
+
+Example:
+- Approximately 30.000 users downloaded and experienced an error using the app.
+- Users already logged in experienced some minor delay in the app
+-->
+
+Team Booking's applications in the development environment went down for approximately five minutes while ArgoCD was
+realizing that there were other applications that had specified the same namespace as well, and therefore brought it
+back up.
+
+## Cause(s)
+<!--
+Describe relevant factors that played a part in causing the event.
+
+Example:
+The error happened due to a combination of:
+- A spike of new users
+- A bug in the backend code for user creation
+-->
+
+### Incident 1
+
+A bug in the ArgoCD app sync upgrade causing the upgrade to ignore what environment the files were related to, migrating
+them all.
```

Comment:
One question remains: Why did we not encounter this issue before this point? Did we only test this upgrade in IAC-repos with one cluster perhaps? If that's the case, I think we should add a comment somewhere in this post mortem about that.